harness/determined/common/api/authentication.py

```diff
@@ -323,7 +271,7 @@ def set_active(self, username: str) -> None:
         with self._persistent_store() as substore:
             tokens = substore.setdefault("tokens", {})
             if username not in tokens:
-                raise api.errors.UnauthenticatedException(username=username)
+                raise api.errors.UnauthenticatedException()
```

Comment:
Not a part of this PR, but I don't get the `cast` in

```
self._active_user = cast(str, substore.get("active_user"))
```

I think it's incorrect. Don't see any bugs, but I think it'd be better to change this anyway.

Reply:
Hm, I think I agree.  I think it should be an `Optional[str]`, not a `str`.

Is that it?  Or do you mean you'd prefer to see something with a runtime check, like:

```python
active_user = substore.get("active_user")
assert isinstance(active_user, (str, type(None))
self._active_user = active_user
```
?

Reply:
Sorry -- I wasn't the most clear. I agree that it should be an `Optional[str]`. The `cast` is making a claim I don't think is accurate.

Reply:
ok, will change to assertion